src/csvcubed/utils/sparql_handler/data_cube_inspector.py

```diff
@@ -188,3 +225,71 @@ def get_code_lists_and_cols(self, csv_url: str) -> CodelistsResult:
         """
 
         return self._codelists_and_cols.get(csv_url, CodelistsResult([], 0))
+
+    def get_column_component_info(self, csv_url: str) -> List[ColumnComponentInfo]:
+        """
+        Get the ColumnComponentInfo which contains the EndUserColumnType
+        and the corresponding ColumnDefinition given a csv url
+        """
+
+        list_to_return = []
+        column_definitions = self.csvw_inspector.get_column_definitions_for_csv(csv_url)
+        column_definitions = [x for x in column_definitions if not x.virtual]
+        qube_components = self.get_dsd_qube_components_for_csv(csv_url).qube_components
+        cube_shape = self.get_shape_for_csv(csv_url)
+
+        observations_columns = {
+            col
+            for comp in qube_components
+            for col in comp.used_by_observed_value_columns
+        }
+
+        def _get_column_type(
+            column: ColumnDefinition,
+        ) -> Tuple[EndUserColumnType, Optional[QubeComponentResult]]:
+            component_definition = first(
+                qube_components, lambda q: column in q.real_columns_used_in
+            )
+
+            if component_definition is None:
+                if column.suppress_output:
+                    return EndUserColumnType.Suppressed, None
+                elif (
+                    cube_shape == CubeShape.Standard and column in observations_columns
+                ):
+                    return EndUserColumnType.Observations, None
+                else:
+                    raise KeyError(
+                        f"Could not find component associated with CSV column '{column.title}'"
+                    )
+
+            return (
+                _figure_out_end_user_column_type(component_definition, cube_shape),
+                component_definition,
+            )
+
+        for column in column_definitions:
+            (column_type, component) = _get_column_type(column)
+            list_to_return.append(
+                ColumnComponentInfo(
+                    component_type=column_type,
+                    column_definition=column,
+                    component=component,
+                )
+            )
+
+        return list_to_return
+
+    def get_columns_for_component_type(
+        self, csv_url: str, column_type: EndUserColumnType
+    ) -> List[ColumnDefinition]:
+
+        list_of_components = self.get_column_component_info(csv_url)
```

Comment:
Can you just call this `components`? And can we just return the following list without having to assign it to a variable?